tests/xharness/Harness.cs

```diff
@@ -657,29 +654,29 @@ public void Log (string message, params object [] args)
 			}
 		}
 
-		public int Execute ()
+		public async Task<int> Execute ()
 		{
 			switch (Action) {
 			case HarnessAction.Configure:
-				return Configure ();
+				return await Configure ();
 			case HarnessAction.Run:
 				return Run ();
 			case HarnessAction.Install:
 				return Install ();
 			case HarnessAction.Uninstall:
 				return Uninstall ();
 			case HarnessAction.Jenkins:
-				return Jenkins ();
+				return await Jenkins ();
 			default:
 				throw new NotImplementedException (Action.ToString ());
 			}
 		}
 
-		int Jenkins ()
+		async Task<int> Jenkins ()
 		{
 			if (autoConf) {
-				AutoConfigureIOS ();
-				AutoConfigureMac (false);
+				await AutoConfigureIOS ();
+				await AutoConfigureMac (false);
```

Comment:
These two lines take 1.8s to execute now.

In `main` they take 0.15s.

This is a significant perf degradation.

Reply:
@rolfbjarne this is a really nice catch. Thanks for that!

I have made one change - the code responsible for building Xamarin.iOS app will stay in this repo as you will be the only users of this code anyway. And you will have it easier to work with it if you won't have to go through `dotnet/xharness`. The templates also diverged already and I think this will be smoother for `xamarin-macios`.

With that I reverted code around these long taking lines so let's see if it worked.

Meanwhile I am setting up new Xcode so that I can debug properly on MacOS.

Reply:
@rolfbjarne I tested locally with the updates/reverts and got this result:
![image](https://user-images.githubusercontent.com/7013027/97847406-d9192f80-1cef-11eb-99c0-fd617dedf1b2.png)

Reply:
@rolfbjarne I tested with the new updates/reverts and got these times back:
![image](https://user-images.githubusercontent.com/7013027/97847666-45942e80-1cf0-11eb-8442-14bc1a4c29db.png)